nservicebus/sagas/concurrency.md

```diff
@@ -1,48 +1,58 @@
 ---
-title: Saga Concurrency
-summary: NServiceBus gives ACID semantics, using underlying storage so only one worker thread hitting a saga instance can commit.
+title: Saga concurrency
+summary: NServiceBus ensures consistency between saga state and messaging.
 component: Core
 tags:
 - Saga
 redirects:
 - nservicebus/nservicebus-sagas-and-concurrency
-reviewed: 2019-02-27
+reviewed: 2019-09-24
 related:
 - persistence/nhibernate/saga-concurrency
 - persistence/ravendb/saga-concurrency
 - persistence/sql/saga-concurrency
 ---
 
-If the endpoint is configured to allow concurrent processing of messages (default) or is scaled out, it is possible that multiple messages will hit the same saga instance simultaneously. To give ACID semantics in this situation, NServiceBus uses the underlying storage to produce consistent behavior, only allowing one of messages to complete. NServiceBus handles most of this automatically but there are some caveats.
+An [endpoint](/nservicebus/concept-overview.md#endpoint) may be configured to allow [concurrent handling of messages](/nservicebus/operations/tuning.md#tuning-concurrency). An endpoint may also be [scaled out](/nservicebus/architecture/scaling.md#scaling-out-to-multiple-nodes) to multiple nodes. In these scenarios, multiple messages may be handled simultaneously by a single logical saga. To ensure consistency between saga state and messages, NServiceBus ensures that only one message handler creates, updates, or deletes saga state in the same transaction as receiving the message and sending new messages. In this document, "handling" a message refers to the invocation of any saga method that processes a message, such as `IAmStartedByMessages<T>.Handle()`, `IHandleTimeouts<T>.Timeout()`, etc.
 
-Concurrent access to saga instances is divided into two scenarios;
+NOTE: When using some transports, the [outbox](/nservicebus/outbox/) must be used in order to provide this level of consistency.
```

Comment:
Some persisters do not support outbox and saga consistency cannot be guaranteed unless the saga state itself has outbox data embedded.

Reply:
@ramonsmits I'm not sure what you mean but this. Let's discuss.